pyam/core.py

```diff
@@ -510,14 +517,125 @@ def check_aggregate(self, variable, components=None, units=None,
                                        **kwargs)]
 
         if len(diff):
-            msg = '{} of {} data points are not aggregates of components'
-            logger().info(msg.format(len(diff), len(df_variable)))
+            msg = '{} - {} of {} data points are not aggregates of components'
+            logger().info(msg.format(variable, len(diff), len(df_variable)))
 
             if exclude_on_fail:
                 self._exclude_on_fail(diff.index.droplevel([2, 3]))
 
+            diff = pd.concat([diff], keys=[variable], names=['variable'])
+
             return diff.unstack().rename_axis(None, axis=1)
 
+    def check_aggregate_regions(self, variable, region='World',
+                                components=None, units=None,
+                                exclude_on_fail=False, multiplier=1, **kwargs):
+        """Check whether the world timeseries data match the aggregation
+        of regions
+
+        Parameters
+        ----------
+        variable: str
+            variable to be checked for matching aggregation of regional data
+        region: str
+            region to be checked for matching aggregation of regional data
+        components: list of str, default None
+            list of regions, defaults to all regions except region
+        units: str or list of str, default None
+            filter variable and components for given unit(s)
+        exclude_on_fail: boolean, default False
+            flag scenarios failing validation as `exclude: True`
+        multiplier: number, default 1
+            factor when comparing variable and sum of components
+        kwargs: passed to `np.isclose()`
+        """
+        # default components to all variables one level below `variable`
+        var_df = self.filter(variable=variable, level=0)
+
+        if components is None:
+            components = var_df.filter(region=region, keep=False).regions()
+
+        if not len(components):
+            msg = (
+                '{} - cannot check regional aggregate because it has no '
+                'regional components'
+            )
+            logger().info(msg.format(variable))
+
+            return None
+
+        # filter and groupby data, use `pd.Series.align` for matching index
+        df_region, df_components = (
+            _aggregate_by_regions(var_df.data, region, units)
+            .align(_aggregate_by_regions(var_df.data, components, units))
+        )
+
+        # if it's a World comparison, we need to add in variables that might
+        # have been missed e.g. shipping and aviation
+        if region == "World":
```

Comment:
I don't think it's a good idea to hard-code any region name, not even ' World' - and I'm not sure we even need this. If we first make sure that all variables aggregate by region, and then check that within each region, they add up to the "parent" variable, the variable only reported at the global level would naturally be included there (and not at the regional level).

Reply:
Nice, will adjust the logic

Reply:
Having re-thought this, here's my use case where it's needed. You want to check that World, Emissions|CO2 is correct. Hence you just add up all the regions for Emissions|CO2. You find that the regional sum is less than World because you're missing variables. I don't see how else to do this ('World' is unique in this respect I think). 

One option would be to make `check_aggregate_regions` a private method which is only used as part of `check_internal_consistency`, in which case your logic would always hold.

Reply:
thanks for setting `World` via a kwarg

Reply:
yes, sorry, I misunderstood this piece of code, I though you had changed this... I think it would be preferably to explicitly state which variables need to be added at the aggregate level (`region='World'`) to get to the sum of components...

Reply:
Ah ok. I did think about doing it that way. The reason I didn't is that it's often hard to predict that a priori. For example, some models have a regional breakdown of aviation (so you don't need to add it again) whilst others don't (so you do need to add aviation in when checking the world total), similarly for aggricultural emissions for CO2. 

Doing it this way ensures that any variables which are in WORLD, but don't have a regional breakdown provided, get included when we check the WORLD total variable. 

I know it's nasty so am happy to clear it up however you think is best.

Reply:
but I'd still think that you could make that logic part of a keyword argument - if you check for a particular region and the subregions (if provided, default all other regions) don't have a particular variable, you include it from the region - otherwise, you aggregate up.

one could think of having World, continents and countries, where some variables are only reported at the continental level. So you would first check that for each continent, internal consistency is given, and then one checks internal consistency from continents to global totals...